### PR TITLE
feat: Show provider logs

### LINKED ISSUE #51: Provide ability to access logs of the provider
### Is your feature request related to a problem? Please describe

When something is going wrong with the podman-machine, it's tedious to understand what's going on.

### Describe the solution you'd like

It should be possible from the provider list to access to the logs of the podman-machine (or the other container engine)

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by evidolob**
Resolves #51

Demo:
Screenshot 2022-04-13 at 11 38 31
Screenshot 2022-04-13 at 11 38 43
Screenshot 2022-04-13 at 11 38 59

>Note: Currently logs from podman machine doesn’t work.

### CODE REVIEW

**Comment by slemeur** [comment]
Nice @evidolob

**Comment by evidolob** [comment]
Podman logs:
Screenshot 2022-04-15 at 10 44 53

File: packages/preload/src/index.ts [hunk 1]
```diff
@@ -310,6 +311,35 @@ function initExtensions(): void {
     return getFreePort(port);
   });
 
+  type logFn = (...data: unknown[]) => void;
+
+  contextBridge.exposeInMainWorld('logs', {
```
Comment (benoitf) [inline packages/preload/src/index.ts]: I think logs is too generic there as it's only for providers

`providerLogs` maybe

File: packages/preload/src/index.ts [hunk 2]
```diff
@@ -310,6 +311,35 @@ function initExtensions(): void {
...
+      log: logFn,
+      warn: logFn,
+      error: logFn,
+      containerConnectionInfo?: ProviderContainerConnectionInfo,
+    ) => {
+      let context;
+      if (containerConnectionInfo) {
+        context = providerRegistry.getMatchingContainerLifecycleContext(providerId, containerConnectionInfo);
+      } else {
+        context = providerRegistry.getMatchingLifecycleContext(providerId);
```
Comment (benoitf) [inline packages/preload/src/index.ts:328]: I think you could have only one method, with containerConnectionInfo being optional parameter

File: packages/renderer/src/app.css [hunk 3]
```diff
@@ -1,3 +1,13 @@
 @tailwind base;
 @tailwind components;
-@tailwind utilities;
\ No newline at end of file
+@tailwind utilities;
+
+.logs {
```
Comment (benoitf) [inline packages/renderer/src/app.css]: We shouldn't have any css there.

File: packages/renderer/src/lib/preferences/PreferencesContainerConnectionRendering.svelte [hunk 4]
```diff
@@ -210,3 +248,17 @@ async function deleteConnection() {
...
+{#if showModal}
+  <Modal
+    on:close="{() => {
+      stopReceivingLogs(showModal);
+      showModal = undefined;
+    }}">
+    <h2 slot="header">Logs</h2>
+    <div id="log" style="height: 400px;">
+      <div style="width:100%; height:100%; display: flex; flexDirection: column;">
+        <textarea class="logs" readOnly name="log" value="{logContent}"></textarea>
```
Comment (benoitf) [inline packages/renderer/src/lib/preferences/PreferencesContainerConnectionRendering.svelte]: I think we should display the logs inside Xterm.js like for container logs.

File: packages/renderer/src/lib/dialogs/Modal.svelte [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+<div class="modal-background" on:click="{close}"></div>
+
+<div class="modal bg-gray-700" role="dialog" aria-modal="true" bind:this="{modal}">
+  <slot name="header" />
+  <hr />
+  <slot />
+  <hr style="margin-bottom: 9px;" />
+
+  <!-- svelte-ignore a11y-autofocus -->
+  <button autofocus on:click="{close}" class="pf-c-button pf-m-primary"> Close </button>
```
Comment (benoitf) [inline packages/renderer/src/lib/dialogs/Modal.svelte]: as the logs seems to be on the full lifecycle of a connection or a Provider I think we need a 'clear' option to prune previous logs

**Comment by benoitf** [comment]
I think we should have a top right `X` icon to close and `close` button should be bottom right aligned or full width.

**Comment by evidolob** [comment]
@benoitf I Update PR, now logs view looks like that:
Screenshot 2022-04-20 at 14 55 48

Could you look again?

**Comment by evidolob** [comment]
@benoitf Logs view also close when user click outside of log view

**Review by benoitf — APPROVED** [review]

**Pull request merged**